src/axom/primal/operators/detail/winding_number_3d_impl.hpp

```diff
@@ -127,16 +132,339 @@ void degenerate_surface_processing(const NURBSPatch<T, 3>& patch,
+      node_dt = rotate_vector_origin(rotator, trimming_curve_data.getQuadratureTangent(q));
+    }
+    else
+    {
+      node = Vector<T, 3>(query, trimming_curve_data.getQuadraturePoint(q));
+      node_dt = Vector<T, 3>(trimming_curve_data.getQuadratureTangent(q));
+    }
+
+    const double node_norm = node.norm();
+    const double quad_weight = trimming_curve_data.getQuadratureWeight(q);
+
+    // Compute one of three vector field line integrals depending on
+    //  the orientation of the original surface, indicated through ax.
+    switch(ax)
+    {
+    case(DiscontinuityAxis::x):
+      this_quad += quad_weight * (node[2] * node[0] * node_dt[1] - node[1] * node[0] * node_dt[2]) /
+        (node[1] * node[1] + node[2] * node[2]) / node_norm;
+      break;
+    case(DiscontinuityAxis::y):
+      this_quad += quad_weight * (node[0] * node[1] * node_dt[2] - node[2] * node[1] * node_dt[0]) /
+        (node[0] * node[0] + node[2] * node[2]) / node_norm;
+      break;
+    case(DiscontinuityAxis::z):
+    case(DiscontinuityAxis::rotated):
+      this_quad += quad_weight * (node[1] * node[2] * node_dt[0] - node[0] * node[2] * node_dt[1]) /
+        (node[0] * node[0] + node[1] * node[1]) / node_norm;
+      break;
+    }
+  }
+
+  return this_quad;
+}
+
+/*!
+ * \brief Evaluates the integral of the "anti-curl" of the GWN integrand
+ *        (via Stokes' theorem) at a point wrt to the trimming curves of a surface
+ *
+ * \param [in] query The query point
+ * \param [in] nurbs The NURBSPatchGWNCache object contianing the trimming curves
+ * \param [in] quad_npts The number of points used in each Gaussian quadrature
+ * \param [in] ax The axis (relative to query) denoting which anti-curl we use
+ * \param [in] rotator This is the rotation matrix to use if ax == DiscontinuityAxis::rotated
+ * \param [in] quad_tol The maximum relative error allowed in each quadrature
+ * 
+ * Applies a non-adaptive quadrature to the trimming curves of a NURBS patch using one of three possible
+ * "anti-curl" vector fields, the curl of each of which is equal to <x, y, z>/||x||^3.
+ *
+ * \return The value of the integral
+ */
+template <typename T>
+double stokes_gwn_evaluate(const Point<T, 3>& query,
+                           const NURBSPatchGWNCache<T>& nurbs,
+                           const int quad_npts,
+                           DiscontinuityAxis ax,
+                           const numerics::Matrix<T>& rotator,
+                           const double quad_tol)
+{
+  // Can't rotate the patch as pre-processing if working with cached data
+  double quad = 0;
+  for(int n = 0; n < nurbs.getNumTrimmingCurves(); ++n)
+  {
+    // Get the quadrature points for the curve on the patch without any refinement
+    auto trimming_curve_data = nurbs.getTrimmingCurveQuadratureData(n, quad_npts, 0, 0);
+    double quad_coarse = stokes_gwn_component(query, ax, rotator, trimming_curve_data);
+
+    quad += 0.25 * M_1_PI *
```

Comment:
The compiler might do this for us, but it might be more efficient to pull the scaling term outside the loop as a `constexpr`,  e.g. 
```cpp
constexpr double sc = 0.25 * M_1_PI;
for(....)
{
   ...
   quad += sc * stokes_gwn_adaptive(...);
}
```